### PR TITLE
Adds components for a selectable table

### PR DESCRIPTION
**Pull request by mko-sci**
### Description

Table when nothing is selected:
Bildschirmfoto 2024-10-22 um 15 43 17

Multiple rows are selected:
Bildschirmfoto 2024-10-22 um 15 43 34

All rows are selected:
Bildschirmfoto 2024-10-22 um 15 43 54

Behavior:
- if the "select-all-checkbox" is being clicked on, all rows are checked.
- if all rows are checked, the "select-all-checkbox" in the `thead` will be checked as well.
- if all rows are checked and the "select-all-checkbox" is checked, all rows will be unselected.
- if all rows are checked and one row gets unchecked, the "select-all-checkbox" is unchecked.
- rows can be checked by either clicking on the checkbox or on the row. 
  - Note: Clicking on links and buttons wont trigger the row selection

### Additional Notes

- This PR fixes or works on following ticket(s): [OX-11093](https://scireum.myjetbrains.com/youtrack/issue/OX-11093)

### Checklist

- [x] Code change has been tested and works locally
- [x] Code was formatted via IntelliJ and follows SonarLint & [best practices](https://scireum.myjetbrains.com/youtrack/articles/MISC-A-16/CodeStyle-JavaDoc)

### CODE REVIEW

**Review by jakobvogel — APPROVED** [review]
The grey of selected lines in the screenshot looks a bit too dark (contrast!)

**Comment by mko-sci** [comment]
> The grey of selected lines in the screenshot looks a bit too dark (contrast!)

Its a bit brighter than the hover effect in reality.

**Comment by sabieber** [comment]
> The grey of selected lines in the screenshot looks a bit too dark (contrast!)

I don't really like the grey in general, as it clashes with the hover effect and the striped-table-mode.
Maybe a very soft blue or some other highlight entirely would be better suited.

File: src/main/resources/default/assets/tycho/scripts/selectableTable.js.pasta [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        _table.querySelectorAll('.select-table-row-checkbox-js').forEach(function (_checkbox) {
+            // Handle changing the state of the select-all checkbox found in the table head row
+            // When all rows are selected, the select-all checkbox should be checked as well
+            _checkbox.addEventListener('change', function () {
+                toggleRowSelection(_checkbox);
+                changeSelectAllCheckboxState(_table);
+            });
+
+            const _tr = _checkbox.closest('tr');
```
Comment (sabieber) [inline src/main/resources/default/assets/tycho/scripts/selectableTable.js.pasta:20]: **Optional:** `_tr` -> `_row`

File: src/main/resources/default/assets/tycho/scripts/selectableTable.js.pasta [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                    _checkbox.checked = !_checkbox.checked;
+                    _checkbox.dispatchEvent(new Event('change'));
+                }
+            });
+
+        });
+    });
+
+    function toggleRowSelection(_checkbox) {
+        const _tr = _checkbox.closest('tr');
```
Comment (sabieber) [inline src/main/resources/default/assets/tycho/scripts/selectableTable.js.pasta:35]: **Optional:** `_tr` -> `_row`

File: src/main/resources/default/assets/tycho/scripts/selectableTable.js.pasta [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        if (_checkbox.checked) {
+            _tr.style.backgroundColor = 'lightgray';
+        } else {
+            _tr.style.backgroundColor = '';
```
Comment (sabieber) [inline src/main/resources/default/assets/tycho/scripts/selectableTable.js.pasta:39]: **Question:** Why not via a color marker class? Also the used color is not really a "sirius" color.

File: src/main/resources/default/assets/tycho/scripts/selectableTable.js.pasta [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    }
+
+    function changeSelectAllCheckboxState(_table) {
+        const allChecked = _table.querySelectorAll('.select-table-row-checkbox-js:checked').length === _table.querySelectorAll('.select-table-row-checkbox-js').length;
+        _table.querySelectorAll('.select-all-visible-table-rows-checkbox-js').forEach(_node => {
```
Comment (sabieber) [inline src/main/resources/default/assets/tycho/scripts/selectableTable.js.pasta:45]: **Optional:** `_node` -> something more descriptive

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Comment by mko-sci** [comment]
bg-sirius-cyan:

Bildschirmfoto 2024-10-23 um 10 19 37

bg-sirius-cyan-light:
Bildschirmfoto 2024-10-23 um 10 24 17

bg-sirius-blue-light:
Bildschirmfoto 2024-10-23 um 10 20 21

bg-sirius-gray-light:
Bildschirmfoto 2024-10-23 um 10 21 57

bg-sirius-gray-lighter:
Bildschirmfoto 2024-10-23 um 10 23 02


IMHO: `bg-sirius-gray-lighter` fits the best!

**Review by scireum-mbo — APPROVED** [review]

**Comment by jakobvogel** [comment]
> IMHO: `bg-sirius-gray-lighter` fits the best!

Hmm, because it is more desaturated than the others. I think @sabieber was referring to pastel-like colors, like the blue used by Apple for text selections, or by MIO for the background of the green conversation tags, etc.

**Review by sabieber — APPROVED** [review]

**Review by jakobvogel — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by jakobvogel — APPROVED** [review]

**Pull request merged**